flexget/plugins/modify/path_select.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+import os
+import random
+from collections import namedtuple
+
+from flexget import plugin
+from flexget.event import event
+from flexget.config_schema import one_or_more
+
+log = logging.getLogger('path_select')
+
+SYMBOLS = {
+    'customary': ('B', 'K', 'M', 'G', 'T', 'P', 'E', 'Z', 'Y'),
+    'customary_ext': ('byte', 'kilo', 'mega', 'giga', 'tera', 'peta', 'exa', 'zetta', 'iotta'),
+    'iec': ('Bi', 'Ki', 'Mi', 'Gi', 'Ti', 'Pi', 'Ei', 'Zi', 'Yi'),
+    'iec_ext': ('byte', 'kibi', 'mebi', 'gibi', 'tebi', 'pebi', 'exbi', 'zebi', 'yobi'),
+}
+
+
+def human2bytes(s):
+    """
+    Attempts to guess the string format based on default symbols
+    set and return the corresponding bytes as an integer.
+    When unable to recognize the format ValueError is raised.
+    """
+
+    init = s
+    num = ""
+    while s and s[0:1].isdigit() or s[0:1] == '.':
+        num += s[0]
+        s = s[1:]
+    num = float(num)
+    letter = s.strip()
+    for name, sset in SYMBOLS.items():
+        if letter in sset:
+            break
+    else:
+        if letter == 'k':
+            # treat 'k' as an alias for 'K' as per: http://goo.gl/kTQMs
+            sset = SYMBOLS['customary']
+            letter = letter.upper()
+        else:
+            raise ValueError("can't interpret %r" % init)
+    prefix = {sset[0]: 1}
+    for i, s in enumerate(sset[1:]):
+        prefix[s] = 1 << (i+1)*10
+    return int(num * prefix[letter])
+
+
+def percentage(part, whole):
+    try:
+        return 100 * part/whole
+    except ZeroDivisionError:
+        return 0.0
+
+
+disk_stats_tuple = namedtuple(
+    'disk_stats', [
+        'path', 'free_bytes', 'used_bytes', 'total_bytes', 'free_percent',  'used_percent'
+    ]
+)
+
+
+def get_disk_stats(folder):
+    """ Return folder/drive stats in megabytes (free, used, total) """
+    if os.name == 'nt':
+        import ctypes
+        free_bytes = ctypes.c_ulonglong(0)
+        total_bytes = ctypes.c_ulonglong(0)
+        ctypes.windll.kernel32.GetDiskFreeSpaceExW(
+            ctypes.c_wchar_p(folder),
+            None,
+            ctypes.pointer(total_bytes),
+            ctypes.pointer(free_bytes)
+        )
+
+        free_bytes = free_bytes.value
+        total_bytes = total_bytes.value
+    else:
+        stats = os.statvfs(folder)
+        free_bytes = stats.f_bavail * stats.f_frsize
+        total_bytes = stats.f_blocks * stats.f_frsize
+
+    free_bytes = int(free_bytes)
+    total_bytes = int(total_bytes)
+    used_bytes = total_bytes - free_bytes
+
+    free_percent = 0.0 if total_bytes == 0 else percentage(free_bytes, total_bytes)
+    used_percent = 0.0 if total_bytes == 0 else percentage(used_bytes, total_bytes)
```

Comment:
I think any added logic should be separate from the function that grabs the actual disk stats from the OS. As is, this logic isn't being tested when running the tests, since the whole function is mocked away. (Granted, it's simple logic in this case, so it probably doesn't matter.) Also, now that you are checking for divide by 0 manually, the `percentage` function seems a bit superfluous.